### PR TITLE
Switch to pydantic for validation and client models

### PR DESCRIPTION
**Pull request by tommyjcarpenter**
- switch to pydantic for validation and client models; eliminate json checking
- add some extra example docs for equity baskets

completes #34 and finishes documenting #41

### CODE REVIEW

File: merqube_client_lib/pydantic_types.py [hunk 1]
```diff
@@ -445,15 +445,22 @@ class PortfolioUom(Enum):
...
+    TICKER = "TICKER"
+    BBG_TICKER = "BBG_TICKER"
+    CUSIP = "CUSIP"
 
 
 class AssetType(Enum):
     CASH = "CASH"
     THIRD_PARTY_INDEX = "THIRD_PARTY_INDEX"
     EQUITY = "EQUITY"
+    MUTUAL_FUND = "MUTUAL_FUND"
```
Comment (tommyjcarpenter) [inline merqube_client_lib/pydantic_types.py:462]: @aschmied these are now available to clients

Comment (ykifle) [reply merqube_client_lib/pydantic_types.py:462]: Is this fully supported yet? I thought there was still more work to be done.

File: merqube_client_lib/templates/equity_baskets/multiple_equity_basket/create.py [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
 from merqube_client_lib.logging import get_module_logger
 from merqube_client_lib.templates.equity_baskets.multiple_equity_basket.base import (
-    TYPE_SPECIFIC_OPTIONAL,
-    TYPE_SPECIFIC_REQUIRED,
     get_constituents,
     inline_to_tp,
 )
+from merqube_client_lib.templates.equity_baskets.schema import (
+    ClientMutliEquityBasketConfig,
```
[the remark this answers is not in the document]
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/multiple_equity_basket/create.py:14]: fixing

File: merqube_client_lib/pydantic_types.py (hunk 1 above)
Comment (tommyjcarpenter) [reply merqube_client_lib/pydantic_types.py:462]: Thats a question for @aschmied - this type file is automatically generated from the API, so they will show up here regardless of backend support.

**Pull request merged**